include/dos_system.h

```diff
@@ -147,7 +147,54 @@ class DOS_File {
 	std::string name = {};
 	FlushTimeOnClose flush_time_on_close = FlushTimeOnClose::NoUpdate;
 	std::vector<FileRegionLock> region_locks = {};
-	/* Some Device Specific Stuff */
+
+	// The MediaDelayGuard class emulates media timing delays for file
+	// operations. It fixes games that either directly measure IO calls or
+	// are sensitive to IO calls that return too quickly. Examples are Deus
+	// (1996), Ishar 3 (1994), Robinson's Requiem (1994), and Time Warriors
+	// (1997). They (otherwise) report an iterrupt #00 handler error, which
+	// is triggered on a divide by zero.
+	//
+	// Create an object in DOS_File derived class's Read, Write, Seek, and
+	// Close functions. Add..() respective quantities. When the function
+	// goes out of scope, the cost for all added quantities will be applied.
+	//
+	class MediaDelayGuard {
+	public:
+		constexpr MediaDelayGuard() = default;
+		~MediaDelayGuard();
+
+		constexpr void AddBytesRead(const uint16_t bytes = 1) noexcept
+		{
+			bytes_read += bytes;
+		}
+		constexpr void AddBytesWritten(const uint16_t bytes = 1) noexcept
+		{
+			bytes_written += bytes;
+		}
+		constexpr void AddSeek() noexcept
+		{
+			++num_seeks;
+		}
+		constexpr void AddVerySlowSeek() noexcept
+		{
+			num_seeks += 50;
+		}
+		constexpr void AddClose() noexcept
+		{
+			++num_closes;
+		}
+		static constexpr void SetDelayPercent(const int scalar) noexcept;
+
+	private:
+		constexpr uint16_t GetNumOperations() const noexcept;
```

Comment:
Just use `int`. Specific-sized integers and especially unsigned types should only ever be used for binary protocols and register-level hackery in our codebase.

I know there was a "push" to start using fixed-size types religiously _everywhere_ in the codebase... but that was a mistake.

More info about it here:
https://google.github.io/styleguide/cppguide.html#Integer_Types